### PR TITLE
bank: Home Page

### PR DESCRIPTION
**Pull request by gadjevb**
Fixes to #57
Initial commit for the servlet responsible for the main page of the bank website!

main

### CODE REVIEW

File: task4/src/main/java/com/clouway/http/servlets/MainPageServlet.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * @author Borislav Gadjev <gadjevb@gmail.com>
+ */
+public class MainPageServlet extends HttpServlet {
+    private String pageForUser;
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response ) throws ServletException, IOException {
+        try {
+            pageForUser = Files.toString(new File("src/main/java/com/clouway/http/resources/index.html"), Charsets.UTF_8);
```
Comment (mgenov) [inline task4/src/main/java/com/clouway/http/servlets/MainPageServlet.java]: Using absolute paths is not a good idea. ClassPath and WebAppContext need to be used for such lookups. 

http://download.eclipse.org/jetty/9.3.11.v20160721/apidocs/org/eclipse/jetty/webapp/WebAppContext.Context.html

File: task4/src/main/test/com/clouway/servlets/MainPageServletTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                oneOf(response).setContentType("text/html");
+                oneOf(response).setStatus(200);
+                oneOf(response).getWriter();
+                will(returnValue(printWriter));
+            }
+        });
+
+        recognizerServlet.doGet(request, response);
+        String page = stringWriter.toString();
+        CharSequence firstExpected = "<title>CTEF Bank</title>";
```
Comment (mgenov) [inline task4/src/main/test/com/clouway/servlets/MainPageServletTest.java]: would be simpler if expectations is on a single line. 

The hamcrest has matcher for this: 
```java
assertThat(page, containsString("<title>CTEF Bank</title>");
```
http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/core/StringContains.html#containsString(java.lang.String)

**Comment by freekman** [comment]
REMOVE bank-skeleton.png

File: task4/src/main/java/com/clouway/http/server/BankServer.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+import javax.servlet.http.HttpServlet;
+
+/**
+ * @author Borislav Gadjev <gadjevb@gmail.com>
+ */
+public class BankServer {
```
Comment (freekman) [inline task4/src/main/java/com/clouway/http/server/BankServer.java]: JettyServer is better

File: task4/src/main/java/com/clouway/http/server/BankServer.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    private final Server server;
+    private final ServletContextHandler context;
+
+    public BankServer(int port, String path){
+        server = new Server(port);
```
Comment (freekman) [inline task4/src/main/java/com/clouway/http/server/BankServer.java]: .

File: task4/src/main/test/com/clouway/servlets/MainPageServletTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    private HttpServletRequest request = context.mock(HttpServletRequest.class);
+    private HttpServletResponse response = context.mock(HttpServletResponse.class);
+
+    @Test
+    public void happyPath() throws ServletException, IOException {
+        MainPageServlet recognizerServlet = new MainPageServlet();
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        context.checking(new Expectations() {
+            {
```
Comment (freekman) [inline task4/src/main/test/com/clouway/servlets/MainPageServletTest.java]: strange formatting

**Comment by freekman** [comment]
@gadjevb  i dont see progres on this PR , i don't see @mgenov's sudgestion to use WebAppContext

**Comment by gadjevb** [comment]
@freekman It will be fixed at later stage.

**Comment by freekman** [comment]
ii think that now is a good time to fix it because now the code is small and the refactor will also be small and further more it will affect your current PR's

File: task4/src/main/java/com/clouway/App.java [hunk 6]
```diff
@@ -1,10 +1,15 @@
...
+
 /**
  * @author Borislav Gadjev <gadjevb@gmail.com>
  */
 public class App {
     public static void main(String[] args) {
-
+        JettyServer server = new JettyServer();
+        server.setPort(8080);
+        server.initHandler(".","src/main/webapp/WEB-INF/web.xml","/");
```
Comment (freekman) [inline task4/src/main/java/com/clouway/App.java]: this can be moved in the JettyServer but leave it like it is if you like it

File: task4/src/main/resources/index.html [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        <p style="text-align: right;">Users online: ${users}</p>
```
Comment (freekman) [inline task4/src/main/resources/index.html]: if this ```Users online: ${users}``` going to be done in a separate PR it has to be removed from this page

**Comment by freekman** [comment]
LGTM after fix comments

**Comment by gadjevb** [comment]
@mgenov i have a LGTM from my mentor but i cant merge because i need a confirm on your change request.

**Comment by mgenov** [comment]
PR is approved.
